src/Sylius/Bundle/OrderBundle/Controller/OrderController.php

```diff
@@ -107,9 +110,16 @@ public function saveAction(Request $request): Response
                 return $this->viewHandler->handle($configuration, View::create(null, Response::HTTP_NO_CONTENT));
             }
 
+            if ($this->isInStock($items, $form)) {
+                $this->addTranslatedFlash('error', 'sylius.cart.quantity_out_of_stock');
+
+                return $this->redirectHandler->redirectToResource($configuration, $resource);
+            }
```

Comment:
I have serious doubts as to whether we should handle it this way, because we do have validation for it, which we are bypassing in this way. Probably we should extend the validation message about the quantity number we wanted to specify and refresh the cart to see the old (saved in db) state of the cart 🤔